infrasim/__init__.py

```diff
@@ -22,5 +22,29 @@ def run_command(cmd="", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PI
     cmd_result = child.communicate()
     cmd_return_code = child.returncode
     if cmd_return_code != 0:
-        return -1, cmd_result[1]
+        result = ""
+        if cmd_result[1] is not None:
+            result = cmd + ":" + cmd_result[1]
+        else:
+            result = cmd
+        logger.error(result)
+        raise CommandRunFailed(result)
```

Comment:
Yes.  This place can grab the full error message and command execute return code